crates/hotshot/src/demo.rs

```diff
@@ -106,7 +104,7 @@ impl TestableState for DemoState {
         _state: Option<&Self>,
         _rng: &mut dyn rand::RngCore,
         padding: u64,
-    ) -> <Self::BlockPayload as BlockPayload>::Transaction {
+    ) -> VIDTransaction {
```

Comment:
Love it!

Reply:
Thanks for the review! I'm actually going to close this PR since we've decided to keep the payload as a trait.